src/platform_impl/macos/app_delegate.rs

```diff
@@ -79,3 +109,36 @@ extern "C" fn did_resign_active(this: &Object, _: Sel, _: id) {
     }
     trace!("Completed `applicationDidResignActive`");
 }
+
+extern "C" fn application_open_file(_this: &Object, _: Sel, _sender: id, filename: id) -> BOOL {
+    trace!("Triggered `application:openFile:`");
+    let string = ns_string_to_rust(filename);
+    let result = AppState::open_files(vec![string.into()]);
+    trace!("Completed `application:openFile:`");
+
+    if result == FileOpenResult::Success {
+        YES
+    } else {
+        NO
+    }
+}
+
+extern "C" fn application_open_files(_this: &Object, _: Sel, sender: id, filenames: id) {
+    trace!("Triggered `application:openFiles:`");
+    let filenames_len = unsafe { filenames.count() };
+    let mut filenames_vec = Vec::with_capacity(filenames_len as usize);
+    for i in 0..filenames_len {
```

Comment:
```rust
let filenames_vec = (0..filenames_len)
    .map(|i| {
        let filename = unsafe { filenames.objectAtIndex(i) };
        ns_string_to_rust(filename).into()
    })
   .collect();
```
IIRC this has the benefits of using `with_capacity` as well as being slightly more concise. Though, iterators still don't seem to be used a ton throughout winit, which is definitely partially my fault (...to be frank, at the peak of my activity, I didn't really know how to use iterators!)

In any case, I'll leave that up to preference as well!

Reply:
What you suggest seems more idiomatic, so I'll go with that.